nix/default.nix

```diff
@@ -2,20 +2,17 @@
 let
   nix-pre-commit-hooks = (import (builtins.fetchTarball "https://github.com/cachix/pre-commit-hooks.nix/tarball/master/" + "/nix/") { sources = sources; }).packages;
   overlay = _self: pkgs:
-    let sharedOverrides = {
-        overrides = _self: super: {
-            mkDerivation = args: super.mkDerivation (args //
-                {
-                    # skip running tests for Hackage packages
-                    doCheck = args.pname == "ghcide" || args.pname == "haskell-language-server";
-                    # relax upper bounds
-                    jailbreak = args.pname != "jailbreak-cabal";
-                });
-            };
-            };
-        gitignoreSource = (import sources.gitignore { inherit (pkgs) lib; }).gitignoreSource;
-        extended = haskellPackages:
-          haskellPackages.extend (pkgs.haskell.lib.packageSourceOverrides {
+    let
+        sharedOverrides = {
+          overrides = _self: super: {
+            mkDerivation = args:
+              let
+                broken = args.broken or false;
+                check = args.doCheck or true;
```

Comment:
Huh, interesting. Now I wonder if we should do the same for the jailbreak variable? Because there are some jailbreaks in nixpkgs. I somehow thought this evaluation happens on the raw package as it is written in the hackage-packages.nix before any overrides have happened. I don‘t think anything is set to dontCheck in there so reading the doCheck wouldn’t be necessary.
otoh if this is **after** overrides we should definitely check and preserve the jailbreak flag, too.

Reply:
Overrides are always a bit magical but it's totally plausible that this is after overrides.

Reply:
But if I remove the check of `args.doCheck`, derivations to be built will get slightly increased.

Reply:
Okay, then you should really do the same for the jailbreak flag!